app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb

```diff
@@ -333,28 +343,36 @@ def find_hypervisor_organization
       deny_access unless @organization
     end
 
-    def find_organization
+    def find_organization(key = :organization_id)
       organization = nil
 
-      if params.key?(:organization_id)
-        organization = Organization.find_by(:label => params[:organization_id])
+      if params.key?(key)
+        organization = Organization.find_by(:label => params[key])
       end
 
       if organization.nil?
         message = _("Couldn't find Organization '%s'.")
-        fail HttpErrors::NotFound, message % params[:organization_id]
+        fail HttpErrors::NotFound, message % params[key]
       end
 
       if User.current && !User.consumer? && !User.current.allowed_organizations.include?(organization)
         message = _("User '%{user}' does not belong to Organization '%{organization}'.")
-        fail HttpErrors::NotFound, message % {:user => current_user.login, :organization => params[:organization_id]}
+        fail HttpErrors::NotFound, message % {:user => current_user.login, :organization => params[key]}
       end
 
       organization
     end
 
+    def convert_organization_label_to_id
+      params[:organization_id] = find_organization.id
+    end
+
+    def convert_owner_to_organization_id
+      params[:organization_id] = find_organization(:owner).id
+    end
+
     def find_activation_keys
-      organization = find_organization
+      organization = Organization.current
```

Comment:
since we loaded the right context by using `additional_set_taxonomy` we now rely on current org here